lib/parser.js

```diff
@@ -192,52 +238,104 @@ function getValidator(version) {
   return validate;
 }
 
-async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
+async function customDocumentOperations(
+  parsedJSON,
+  asyncapiYAMLorJSON,
+  initialFormat,
+  options
+) {
   validateServerVariables(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateServerSecurity(parsedJSON, asyncapiYAMLorJSON, initialFormat, SPECIAL_SECURITY_TYPES);
+  validateServerSecurity(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    SPECIAL_SECURITY_TYPES
+  );
 
   if (!parsedJSON.channels) return;
 
   validateTags(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
-  validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
-
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  validateOperationId(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    OPERATIONS
+  );
+
+  await customComponentsMsgOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
+  await customChannelsOperations(
+    parsedJSON,
+    asyncapiYAMLorJSON,
+    initialFormat,
+    options
+  );
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(
+  msg,
+  originalAsyncAPIDocument,
+  fileFormat,
+  parsedAsyncAPIDocument,
+  pathToPayload
+) {
   //check if the message has been parsed before
-  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
-  const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
+  if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true)
+    return;
+  const defaultSchemaFormat = getDefaultSchemaFormat(
+    parsedAsyncAPIDocument.asyncapi
+  );
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
 
-  await PARSERS[String(schemaFormat)]({
+  const parser = PARSERS[String(schemaFormat)];
+
+  if (!parser) {
+    throw new ParserError({
+      type: 'schema-parser-not-registered',
+      title: 'The specification has an unsupported schema format',
+      detail:
+        'You shall not pass, no schema parser available that can parse this schema. The good news is that we have one for Avro, just install @asyncapi/avro-schema-parser and use registerSchemaParser to register it with the current parser',
+      validationErrors: groupValidationErrors(
+        'servers',
+        'server does not have a corresponding schema object',
+        new Map(),
+        fileFormat,
+        parser
```

Comment:
here you should pass information if original AsyncAPI file passed into parser was YAML or JSON. It is in `fileFormat` variable.